### PR TITLE
[AUD-1797] Add CreatePlaylistScreen

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Adds CreatePlaylistScreen
- Refactors EditPlaylistScreen and reuses input components in CreatePlaylistScreen
- Adds disabled, icon and icon placement to TextButton
- Adds required/ validation for input-field
- Adds FormScreen and uses it for all our form screens

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1797-add-create-playlist

File: packages/mobile/src/components/core/TextButton.tsx [hunk 1]
```diff
@@ -34,14 +38,42 @@ const useStyles = makeStyles(({ typography, palette }, { variant }) => {
...
+      width={18}
+      fill={styles.icon.fill}
+      style={iconPosition === 'left' ? styles.iconLeft : styles.iconRight}
+    />
+  ) : null
+
   return (
-    <TouchableOpacity {...other}>
-      <Text style={styles.text}>{title}</Text>
+    <TouchableOpacity
```
Comment (sliptype) [inline packages/mobile/src/components/core/TextButton.tsx:67]: Almost seems like TextButton should just be a variant of our core Button

**Review by sliptype — APPROVED** [review]
Nice! Thanks for breaking things into separate components

File: packages/mobile/src/components/core/TextButton.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/mobile/src/components/core/TextButton.tsx:67]: yeah it definitely agreed, prop talk with julian about it when design system becomes more of a thing

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-aud-1797-add-create-playlist

**Pull request merged**